app-modules/meeting-center/src/GoogleCalendarManager.php

```diff
@@ -0,0 +1,205 @@
+
+        $events = collect();
+
+        do {
+            $googleEvents = $service->events->listEvents($calendar->provider_id, $parameters);
+            $parameters['pageToken'] = $googleEvents->nextPageToken;
+
+            $events = $events->merge($service->events->listEvents($calendar->provider_id, $parameters)->getItems());
+        } while ($googleEvents->nextPageToken);
+
+        return $events->toArray();
+    }
+
+    public function createEvent(CalendarEvent $event): CalendarEvent
+    {
+        /**
+         * Warning: If you add an event using the values declined, tentative, or accepted, attendees
+         * with the "Add invitations to my calendar" setting set to "When I respond to invitation in email"
+         * won't see an event on their calendar unless they choose to change their invitation response in
+         * the event invitation email.
+         *
+         * @see https://developers.google.com/calendar/api/v3/reference/events/insert
+         *
+         * @todo check for overlapping events
+         * @todo add attendees?
+         * @todo auto accept?
+         * @todo create later?
+         * */
+        $service = (new GoogleCalendar(static::client($event->calendar)));
+
+        $googleEvent = $this->toGoogleEvent($event);
+        $googleEvent = $service->events->insert($event->calendar->provider_id, $googleEvent);
+        $event->provider_id = $googleEvent->getId();
+
+        return $event;
+    }
+
+    public function updateEvent(CalendarEvent $event): CalendarEvent
+    {
+        if ($event->provider_id) {
+            $googleEvent = $this->toGoogleEvent($event);
+            $service = (new GoogleCalendar(static::client($event->calendar)));
+            $service->events->update($event->calendar->provider_id, $event->provider_id, $googleEvent);
+        } else {
+            $event = $this->createEvent($event);
+        }
+
+        return $event;
+    }
+
+    public function deleteEvent(CalendarEvent $event): void
+    {
+        $service = (new GoogleCalendar(static::client($event->calendar)));
+        $service->events->delete($event->calendar->provider_id, $event->provider_id);
+    }
+
+    public function syncEvents(Calendar $calendar): void
+    {
+        $events = collect($this->getEvents($calendar));
+
+        $events
+            ->each(
+                function (Event $event) use ($calendar) {
+                    $userEvent = $calendar->events()->where('provider_id', $event->id)->first();
+
+                    if ($userEvent) {
+                        $userEvent
+                            ->updateQuietly([
+                                'title' => $event->summary,
+                                'description' => $event->description,
+                                'starts_at' => $event->start->dateTime,
+                                'ends_at' => $event->end->dateTime,
+                            ]);
```

Comment:
Will this still trigger a DB query if there is no change?

If not, we may want to consider filling, and then only triggering the update if dirty.